src/main/java/net/mcreator/ui/dialogs/NewVariableDialog.java

```diff
@@ -41,7 +42,7 @@ public static VariableElement showNewVariableDialog(JFrame frame, boolean showSc
 		variableNameValidator.setValidatedComponent(textField);
 		textField.setValidator(variableNameValidator);
 
-		JComboBox<VariableElementType> type = new JComboBox<>(supportedTypes);
+		JComboBox<String> type = new JComboBox<>(supportedTypes);

```

Comment:
I suggest you keep using variable object here and not string, applies to most other places too. It should implement toString to provide string value, but I see no need in refactoring to String as you did in most places.